instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/v0_11/KafkaSingletons.java

```diff
@@ -72,22 +69,28 @@ public static Instrumenter<KafkaProcessRequest, Void> consumerProcessInstrumente
     return CONSUMER_PROCESS_INSTRUMENTER;
   }
 
+  @SuppressWarnings("unchecked")
   public static void enhanceConfig(Map<? super String, Object> config) {
-    if (!METRICS_ENABLED) {
+    if (!METRICS_ENABLED
+        || config.get(OpenTelemetryMetricsReporter.CONFIG_KEY_OPENTELEMETRY_INSTRUMENTATION_NAME)
+            != null) {
       return;
     }
     config.merge(
         CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
         OpenTelemetryMetricsReporter.class.getName(),
         (class1, class2) -> {
-          if (class1 instanceof String) {
+          // class1 is either a class name or List of class names or classes
+          if (class1 instanceof List) {
+            List<Object> result = new ArrayList<>();
+            result.addAll((List<Object>) class1);
+            result.add(class2);
+            return result;
+          } else if (class1 instanceof String) {
             String className1 = (String) class1;
             if (className1.isEmpty()) {
               return class2;
             }
-            if (METRIC_REPORTER_PRESENT_PATTERN.matcher(className1).find()) {
-              return class1;
-            }
```

Comment:
Not 100% sure of the original intent, but it sure looked like the pattern was here to attempt to prevent duplicates. Removing this now allows duplicates, which could possibly allow for duplicated telemetry (unless there's some other hidden guardrails in place).

In any case, this is a change in behavior. Looks like there wasn't test coverage anyway, so .........

Reply:
There is a test for this. Duplicates are now handled with `config.get(OpenTelemetryMetricsReporter.CONFIG_KEY_OPENTELEMETRY_INSTRUMENTATION_NAME) != null` check at the start of the method as outlined in the pr description